Build error with intel compiler & spack

I've built openblas with spack using intel@18.0.3 and when I try to run a simple program it gives the following errors: 
```
PATH_TO_OPENBLAS/lib/libopenblas.so: undefined reference to `for_cpystr'
PATH_TO_OPENBLAS/lib/libopenblas.so: undefined reference to `for_concat'
```
Here's how the linking looks like :
```
[sajid@xrmlite lib]$ pwd
/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/intel-18.0.3/openblas-0.3.4-meisjl5yp4xcjywvq3melryeyueubox2/lib
[sajid@xrmlite lib]$ ldd libopenblas.so
        linux-vdso.so.1 =>  (0x00007ffe29283000)
        libm.so.6 => /lib64/libm.so.6 (0x00002b1aa8aac000)
        libimf.so => /opt/intel/compilers_and_libraries_2018.3.222/linux/compiler/lib/intel64/libimf.so (0x00002b1aa8dae000)
        libsvml.so => /opt/intel/compilers_and_libraries_2018.3.222/linux/compiler/lib/intel64/libsvml.so (0x00002b1aa9341000)
        libirng.so => /opt/intel/compilers_and_libraries_2018.3.222/linux/compiler/lib/intel64/libirng.so (0x00002b1aaac50000)
        libgcc_s.so.1 => /home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-7.3.0-xyzezhjmbiebkjfoakso464rhfshlkyq/lib64/libgcc_s.so.1 (0x00002b1aaafc4000)
        libintlc.so.5 => /opt/intel/compilers_and_libraries_2018.3.222/linux/compiler/lib/intel64/libintlc.so.5 (0x00002b1aab1db000)
        libc.so.6 => /lib64/libc.so.6 (0x00002b1aab44a000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00002b1aab817000)
        /lib64/ld-linux-x86-64.so.2 (0x000055ba20f17000)
```
And for reference, here is the build log : [[build.out](https://transfer.sh/gcn3K/build.out)] and the [[build.env](https://transfer.sh/MXSVn/build.env.redacted)]. 
spack version: develop, at the latest commit.

--- a/exports/Makefile
+++ b/exports/Makefile
@@ -141,6 +141,14 @@ else
 	$(OBJCOPY) --redefine-syms objcopy.def ../$(LIBNAME) ../$(LIBNAME).renamed
 ../$(LIBSONAME) : ../$(LIBNAME).renamed linktest.c
 endif
+
+ifeq ($(F_COMPILER), INTEL)
+	$(FC) $(FFLAGS) $(LDFLAGS) -shared -o ../$(LIBSONAME) \
+	-Wl,--whole-archive $< -Wl,--no-whole-archive \
+	-Wl,-soname,$(INTERNALNAME) $(EXTRALIB)
+	$(CC) $(CFLAGS) $(LDFLAGS) -w -o linktest linktest.c ../$(LIBSONAME) $(FEXTRALIB) && echo OK.
+else
+
 ifneq ($(C_COMPILER), LSB)
 	$(CC) $(CFLAGS) $(LDFLAGS) -shared -o ../$(LIBSONAME) \
 	-Wl,--whole-archive $< -Wl,--no-whole-archive \
@@ -152,6 +160,7 @@ else
 	-Wl,--whole-archive $< -Wl,--no-whole-archive \
 	-Wl,-soname,$(INTERNALNAME) $(EXTRALIB)
 	$(FC) $(CFLAGS) $(LDFLAGS) -w -o linktest linktest.c ../$(LIBSONAME) $(FEXTRALIB) && echo OK.
+endif
 endif
 	rm -f linktest
 

